edubusluc/Artists-Haven
# Feature/r 001 user registration

### Description

_Provide a brief but clear description of the purpose of this Pull Request (PR). Include the main objective, the changes made, and any important details for review._

### Type of change

- [x] **New feature (feat)**: A new feature is added.
- [ ] **Bug fix (fix)**: Issues or bugs are resolved.
- [ ] **Code refactor (refactor)**: Changes in the code that do not affect functionality.
- [ ] **Documentation (docs)**: Changes to the project documentation.
- [x] **Tests (test)**: Tests are added or modified.
- [ ] **Other**: _(please specify if necessary)_

### Expected behavior

- User registration:
     - Successful registration with valid, unique email, name, and password should create a user and return `201 Created`.
     - If the email already exists, return 400 Bad Request with an appropriate error message.

- Artist registration:
     - Successful registration with valid, unique email, artist name, URL, and password should create an artist and return `201 Created`.
     - If the artist name or email already exists, return 400 Bad Request with an error message.

### Changes made

- [x] Created Artist and User entity.
- [x] Implemented user registration functionality for both normal users and artists.
- [x] Developed unit tests for the user and artist registration process, ensuring that both functionalities work as expected

### How to test the changes?

- Steps to test the new functionality.
             1. Ensure that the application is running and the database is correctly configured.
             2. Test the user registration for both normal users and artists by making POST requests to the /register endpoint with valid data for both types of users.
             3. Verify that the response returns the correct status (201 Created for success, 400 Bad Request for validation errors).
             4. Check that the proper validation messages are displayed when invalid data (e.g., duplicate email or artist name) is submitted.

- Are there unit or integration tests? If so, how to run them?
    - Yes, unit tests have been written for both the user and artist registration functionalities.
    - To run the tests, use the following Maven command: `./mvnw test`

### Checklist

- [x] The code follows the project's conventions (method names, coding style, etc.).
- [x] I have performed local tests before creating the PR.
- [x] Tests are complete and working correctly.
- [x] I have tested the functionality in different browsers/devices (if applicable).

### Related Issue(s)
- Closes #1
